src/app/shared/model/campaign.model.ts

```diff
@@ -0,0 +1,63 @@
+import { BaseModel, HalDoc } from 'ngx-prx-styleguide';
+import { Observable } from 'rxjs/Observable';
+import { SponsorModel } from './sponsor.model';
+
+export class CampaignModel extends BaseModel {
+  public id: number;
+  public startDate: Date;
+  public endDate: Date;
+  public dueDate: Date;
+  public copy: string;
+  public zone: string;
+  public approved = false;
+  public sponsor: SponsorModel;
+
+  SETABLE = ['copy', 'approved'];
+
+  VALIDATORS = {};
+
+  constructor(podcast: HalDoc, campaign: HalDoc, loadRelated = true) {
+    super();
+    this.init(podcast, campaign, loadRelated);
+  }
+
+  key(): string {
+    if (this.doc) {
+      return `prx.campaign.${this.doc.id}`;
+    }
+  }
+
+  related(): {} {
+    let sponsor = Observable.of(null);
+    if (this.doc && this.doc.has('prx:sponsor')) {
+      sponsor = this.doc.follow('prx:sponsor').map(sDoc => new SponsorModel(sDoc));
+    }
+    return { sponsor: sponsor };
+  }
+
+  decode(): void {
+    this.id = this.doc['id'];
+    this.startDate = new Date(this.doc['start_date']);
```

Comment:
I'm curious about the snake_case vs camelCase on these fields. Do they come in with snake_case and go out with camelCase?

Reply:
Interesting, this is making me realize that CMS and Feeder have their API in camelCase. Will add a ticket to make Jingle do the same, and assume they come in camelCase here! Good catch. 
jingle ticket: https://github.com/PRX/jingle.prx.org/issues/66